replica-node/src/main.rs

```diff
@@ -0,0 +1,356 @@
+    );
+
+    let ledger_path = PathBuf::from(matches.value_of("ledger_path").unwrap());
+    let snapshot_output_dir = if matches.is_present("snapshots") {
+        PathBuf::from(matches.value_of("snapshots").unwrap())
+    } else {
+        ledger_path.clone()
+    };
+    let snapshot_path = snapshot_output_dir.join("snapshot");
+
+    let account_paths: Vec<PathBuf> =
+        if let Ok(account_paths) = values_t!(matches, "account_paths", String) {
+            account_paths
+                .join(",")
+                .split(',')
+                .map(PathBuf::from)
+                .collect()
+        } else {
+            vec![ledger_path.join("accounts")]
+        };
+
+    let rpc_source_addr =
+        solana_net_utils::parse_host_port(matches.value_of("peer").unwrap_or_else(|| {
+            clap::Error::with_description(
+                "The --peer <IP:PORT> argument is required",
+                clap::ErrorKind::ArgumentNotFound,
+            )
+            .exit();
+        }))
+        .unwrap_or_else(|e| {
+            eprintln!("failed to parse entrypoint address: {}", e);
+            exit(1);
+        });
+
+    let rpc_port = value_t!(matches, "rpc_port", u16).unwrap_or_else(|_| {
+        clap::Error::with_description(
+            "The --rpc-port <PORT> argument is required",
+            clap::ErrorKind::ArgumentNotFound,
+        )
+        .exit();
+    });
+    let rpc_addrs = (
+        SocketAddr::new(rpc_bind_address, rpc_port),
+        SocketAddr::new(rpc_bind_address, rpc_port + 1),
+        // If additional ports are added, +2 needs to be skipped to avoid a conflict with
+        // the websocket port (which is +2) in web3.js This odd port shifting is tracked at
+        // https://github.com/solana-labs/solana/issues/12250
+    );
+
+    let logfile = {
+        let logfile = matches
+            .value_of("logfile")
+            .map(|s| s.into())
+            .unwrap_or_else(|| format!("solana-replica-node-{}.log", identity_keypair.pubkey()));
+
+        if logfile == "-" {
+            None
+        } else {
+            println!("log file: {}", logfile);
+            Some(logfile)
+        }
+    };
+
+    let _logger_thread = solana_validator::redirect_stderr_to_file(logfile);
+
+    let (cluster_info, rpc_contact_info, snapshot_info) = replica_util::get_rpc_peer_info(
+        identity_keypair,
+        &cluster_entrypoints,
+        &ledger_path,
+        &node,
+        expected_shred_version,
+        &peer_pubkey,
+        &snapshot_output_dir,
+    );
+
+    info!(
+        "Using RPC service from node {}: {:?}, snapshot_info: {:?}",
+        rpc_contact_info.id, rpc_contact_info.rpc, snapshot_info
+    );
+
+    let config = ReplicaNodeConfig {
+        rpc_source_addr,
+        rpc_addr: rpc_addrs.0,
+        rpc_pubsub_addr: rpc_addrs.1,
+        ledger_path,
+        snapshot_output_dir,
+        snapshot_path,
+        account_paths,
+        snapshot_info: snapshot_info.unwrap(),
```

Comment:
Is this unwrap safe?

Reply:
Bump

Reply:
As the code written yes. I would have exited before otherwise.